app/stats/potion.py

```diff
@@ -0,0 +1,34 @@
+class Potion:
+    def __init__(
+            self,
+            name: str,
+            protection: int = 0,
+            hp: int = 0,
+            power: int = 0
+    ) -> None:
+        self.name = name
+        self.protection = protection
+        self.hp = hp
+        self.power = power
+
+    @staticmethod
+    def potion_registration(knights_potion: dict = None) -> "Potion" or None:
```

Comment:
u can't use "or" in the typehint like "Potion" or None; use Union[..., ...] or | (Pipe)

use typehint like: Optional[YOUR_VALUE_TYPE] = None
for example: list that contains string elements and can be None has typehint Optional[list[str]]